### ISSUE TITLE
AWS SageMakerのハイパーパラメータ調整ジョブ内Environment値が保護されない

### ISSUE THREAD

**Issue by jyup-escape** [EdamAme-x/pentect#785]
# SEC-345 AWS SageMakerのハイパーパラメータ調整ジョブ内`Environment`値が保護されない

## 概要

AWS CLIの`aws sagemaker create-hyper-parameter-tuning-job`は、複数の学習ジョブを起動してハイパーパラメータを探索する操作です。AWS公式資料では、`TrainingJobDefinition`内の`Environment`を、調整ジョブが起動する学習ジョブへ渡す環境変数マップとして定義しています。複数の学習ジョブ定義を指定できるため、同じ認証値が調整処理の設定から各学習ジョブへ伝播する入力経路になります。

公式構造に沿った`--training-job-definitions`のJSON入力へ合成値を配置してPentectの`mask`処理へ渡したところ、環境変数値が平文のまま出力され、診断は`masked 0 value(s), 0 warned`となりました。`DB_URL`のような名前と、`FIRST`・`SECOND`のような任意名の複数値の両方で確認しています。

AWS公式資料は、この環境変数が`CreateTrainingJob` APIへ渡されることを説明しています。したがって、調整ジョブの開始入力に含まれる値は、最終的に起動される学習コンテナーの環境変数となり得ます。外部サービスの接続情報、API鍵、署名用秘密などをこの欄へ置く利用形態では、平文のまま後続処理へ渡る可能性があります。

## 影響範囲

- `aws sagemaker create-hyper-parameter-tuning-job --training-job-definitions`
- `TrainingJobDefinition.Environment`のJSONマップ
- ハイパーパラメータ調整ジョブから起動される学習ジョブ
- 学習コンテナーの外部サービス接続設定
- 端末記録、シェル履歴、調整ジョブ起動ログ、継続的インテグレーション記録
- AIエージェントへ渡すAWS CLI入力

調整ジョブは、設定した学習ジョブ定義を複数回利用するため、設定値の露出範囲が単一の学習ジョブ開始より広がります。Pentectがこの入力を保護しない場合、起動前の端末出力や自動化記録へ平文が残るほか、起動後の複数の学習コンテナーへ同じ値が渡る構成を見落とす可能性があります。

## 再現手順

実在のAWSアカウント、SageMakerジョブ、IAM認証情報、S3バケット、外部通信は使っていません。合成値を含むAWS CLI文字列を、作業ツリーで再構築したPentect実行ファイルの`mask`処理へ標準入力から渡しました。AWS CLI、SageMaker API、ハイパーパラメータ調整ジョブ開始は実行していません。

### `TrainingJobDefinition.Environment`へ単一値を指定する場合

```powershell
$exe = 'D:\pentect-target-bugbounty-20260825\debug\pentect.exe'
$input = 'aws sagemaker create-hyper-parameter-tuning-job --hyper-parameter-tuning-job-name synthetic-tuning --training-job-definitions [{"DefinitionName":"synthetic-definition","Environment":{"DB_URL":"SageMakerTuningDb987!"}}]'
$input | & $exe mask
```

実際の出力:

```text
aws sagemaker create-hyper-parameter-tuning-job --hyper-parameter-tuning-job-name synthetic-tuning --training-job-definitions [{"DefinitionName":"synthetic-definition","Environment":{"DB_URL":"SageMakerTuningDb987!"}}]
[pentect] profile=Strict masked 0 value(s), 0 warned.
```

`TrainingJobDefinition.Environment.DB_URL`の合成値が平文のまま残りました。

### 任意名の複数値を指定する場合

```text
aws sagemaker create-hyper-parameter-tuning-job --hyper-parameter-tuning-job-name synthetic-tuning --training-job-definitions [{"DefinitionName":"synthetic-definition","Environment":{"FIRST":"SageMakerTuningFirst987!","SECOND":"SageMakerTuningSecond987!"}}]
[pentect] profile=Strict masked 0 value(s), 0 warned.
```

環境変数名を`FIRST`、`SECOND`へ変えても、両方の値が平文で出力されました。調整ジョブの学習ジョブ定義にある環境変数マップが、JSONの入れ子構造として走査されていません。

### 期待される変換例

```text
aws sagemaker create-hyper-parameter-tuning-job --hyper-parameter-tuning-job-name synthetic-tuning --training-job-definitions [{"DefinitionName":"synthetic-definition","Environment":{"DB_URL":"PROTECTED_HANDLE"}}]
[pentect] profile=Strict masked 1 value(s), 0 warned.
```

保護ハンドルは説明用の表記です。実行結果から取得したハンドルや実在の秘密値を報告本文へ保存していません。

## 公式仕様との対応

[AWS公式の`aws sagemaker create-hyper-parameter-tuning-job`資料](https://docs.aws.amazon.com/cli/latest/reference/sagemaker/create-hyper-parameter-tuning-job.html)は、`--training-job-definitions`を調整ジョブから起動する`HyperParameterTrainingJobDefinition`の一覧として定義しています。同資料の各定義には`Environment`マップがあり、そこへ指定した環境変数が`CreateTrainingJob` APIへ渡されると説明されています。

同資料は、`Environment`のキーと値の制約、JSON構文`"Environment": {"string": "string"}`、および調整ジョブ自体と各学習ジョブ定義における環境変数数の扱いを記載しています。今回の入力は、この公式構造の`Environment`へ合成値を配置したものです。

## 期待される動作

`create-hyper-parameter-tuning-job`の`--training-job-definitions`を認識し、各定義の`Environment`マップ内にある値本体を保護することが期待されます。`DB_URL`のような名前や任意名の環境変数、単一値や複数値、複数の学習ジョブ定義を含む場合でも、キー名や学習ジョブ設定と値本体を区別して処理する必要があります。

## 実装上の根拠

AWS公式資料にある`TrainingJobDefinition.Environment`のJSON構造を含む入力で、単一の`DB_URL`値と任意名の二つの値がともに`masked 0`となりました。AWSへ接続していないため、影響確認はCLI入力を保護するPentectの検査結果に限定していますが、公式の調整ジョブ構造へ対応する入力経路で環境変数の保護処理が適用されていないことを再現できます。

## 確認状況

- 使用した値: `SageMakerTuningDb987!`、`SageMakerTuningFirst987!`、`SageMakerTuningSecond987!`
- 実在の認証情報: 使用していません
- AWSへの接続: 実施していません
- Pentect実行結果: JSONの単一値・複数値で`masked 0`、`warned 0`
- 公開本文へ記載した値: すべて合成値
- 期待される保護後表記: `PROTECTED_HANDLE`
